packages/jasmine-parameterized/src/parameterize.ts

```diff
@@ -0,0 +1,83 @@
+import { Spec, SpecOverride, Suite, SuiteOverride } from './types.js';
+
+/**
+ * Used to create a parameterized test or suite using an object of names and arbitrary test values.
+ * In the following example:
+ *  - the test named `catsAndDogs` is focused for debugging
+ *  - the test named `frogs` is configured to always be disabled
+ *  - the test named `men` will run normally as it has no override
+ * @example
+ * const rainTests = {
+ *     catsAndDogs: 'idiom',
+ *     frogs: 'idiom',
+ *     men: 'lyrics'
+ * } as const;
+ * describe('Different rains', () => {
+ *     parameterize(rainTests, (spec, name, value) => {
+ *         spec(`of type ${name} exist`, () => {
+ *             expect(value).toBeDefined();
+ *         });
+ *     }, it, {
+ *         catsAndDogs: fit,
+ *         frogs: xit
+ *     });
+ * });
+ */
+export function parameterize<T extends object>(
+    testCases: T,
+    test: (spec: Spec, name: keyof T, value: T[keyof T]) => void,
+    defaultTestType: Spec,
+    overrides?: {
+        [P in keyof T]?: SpecOverride;
+    }
+): void;
+export function parameterize<T extends object>(
+    testCases: T,
+    test: (spec: Suite, name: keyof T, value: T[keyof T]) => void,
+    defaultTestType: Suite,
+    overrides?: {
+        [P in keyof T]?: SuiteOverride;
+    }
+): void;
+export function parameterize<T extends object, U = Spec | Suite>(
+    testCases: T,
+    test: (spec: U, name: keyof T, value: T[keyof T]) => void,
+    defaultTestType: U,
+    overrides?: {
+        [P in keyof T]?: U extends Spec ? SpecOverride : SuiteOverride;
+    }
+): void {
+    const testCaseNames = Object.keys(testCases) as (keyof T)[];
+    if (overrides) {
+        const overrideNames = Object.keys(
+            overrides
+        ) as (keyof typeof overrides)[];
+        if (
+            !overrideNames.every(overrideName => testCaseNames.includes(overrideName))
+        ) {
+            throw new Error(
+                'Parameterized test override names must match test case name'
+            );
+        }
+        if (
+            defaultTestType === it
```

Comment:
nit: It's a little weird that we're assuming `defaultTestType` is either `it` or `describe` (i.e. never `fit`, `xit`, `fdescribe`, or `xdescribe`). I spent a while trying to figure out a way to let the function use either `it` or `describe` as needed (instead of passing in a `defaultTestType`), but I guess you just can't do that without run-time types.

Reply:
I wasn't a huge fan of that either, but I thought the trade-off of being able to single-source this function for specs & suites was worth that concession, especially since this is an internal function. However, it would probably make sense to throw an error if the default type isn't either `it` or `describe` since we are making that assumption within the code.

Reply:
I ended up updating this in a different way based on another review discussion.